### ISSUE TITLE
Confirmation of behavior upon a failure of a service that depends on another one

### ISSUE THREAD

**Issue by pypingou** [engelmi/hirte#619]
## Please describe what you would like to see

I'd like confirmation that have a test covering this use-case and if not we should add it to our test matrix.
Here is the use-case:

- A on node a requires B on node b
- Service A fails
- Systemd is configured to restart A and B upon failure of A
- Ensure that B on b is also restarted when A crashes on a

This should be consistent with how systemd works on a system when a service A crashes and is restarted by systemd (if configured accordingly)

**Comment by engelmi** [comment]
I assume that the starting order is first B(b) then A(a) since A(a) requires B(b), right? 
There is currently no such test case, but it should be possible to create one.

**Comment by engelmi** [comment]
After a bit of investigation and extending our proxy service documentation, I found a few [limitations](https://bluechi.readthedocs.io/en/latest/cross_node_dependencies/proxy_services/#limitations) of the cross-node dependency feature compared to local systemd. I also added a few use case examples and how to solve them via the proxy services in a [usage section](https://bluechi.readthedocs.io/en/latest/cross_node_dependencies/usage/) which we can extend in the future. 

Basically, the described scenario above could be implemented with [PartOf=](https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#PartOf=) on service B - so restarting of A also restarts B. But since PartOf= is not supported by BlueChi (see limitations in docs), this can't be achieved with BlueChi. 

I don't know the use case, but assuming the situation where service A needs service B and when starting service A fails, but B doesn't. I'd consider restarting B as well an edge case. And if B fails, it can be restarted by A using `Upholds=`, for example. I'll add an integration test for that.
